osu.Game.Rulesets.Osu/Difficulty/Skills/Aim.cs

```diff
@@ -14,51 +16,115 @@ namespace osu.Game.Rulesets.Osu.Difficulty.Skills
     /// </summary>
     public class Aim : OsuStrainSkill
     {
-        private const double angle_bonus_begin = Math.PI / 3;
-        private const double timing_threshold = 107;
-
         public Aim(Mod[] mods)
             : base(mods)
         {
         }
 
-        protected override double SkillMultiplier => 26.25;
+        protected override int HistoryLength => 2;
+
+        protected override double SkillMultiplier => 24.75;
         protected override double StrainDecayBase => 0.15;
 
+        private const double wide_angle_multiplier = 1.0;
+        private const double acute_angle_multiplier = 1.0;
+        private const double rhythm_variance_multiplier = 1.0;
+        private const double slider_multiplier = 6.5;
+        private const double slider_jump_multiplier = 0.875;
+        private const double vel_change_multiplier = 6.5;
+
         protected override double StrainValueOf(DifficultyHitObject current)
         {
-            if (current.BaseObject is Spinner)
+            if (current.BaseObject is Spinner || Previous.Count <= 1)
                 return 0;
 
-            var osuCurrent = (OsuDifficultyHitObject)current;
+            var osuCurrObj = (OsuDifficultyHitObject)current;
+            var osuPrevObj = (OsuDifficultyHitObject)Previous[0];
+            var osuLastObj = (OsuDifficultyHitObject)Previous[1];
+
+            var currVector = Vector2.Divide(osuCurrObj.JumpVector, (float)osuCurrObj.StrainTime);
+            var prevVector = Vector2.Divide(osuPrevObj.JumpVector, (float)osuPrevObj.StrainTime);
```

Comment:
These could be renamed to `currVelocity` and `prevVelocity` for more clarity

Reply:
will let this be handled by further reviews under https://github.com/ppy/osu/pull/14845